src/lib/operators/insert.cpp

```diff
@@ -119,9 +119,18 @@ std::shared_ptr<const Table> Insert::_on_execute(std::shared_ptr<TransactionCont
           ChunkRange{target_chunk_id, target_chunk->size(),
                      static_cast<ChunkOffset>(target_chunk->size() + num_rows_for_target_chunk)});
 
+      // Grow MVCC vectors and mark new (but still empty) rows as being under modification by current transaction.
+      // Do so before resizing the Segments, because the resize of `Chunk::_segments.front()` is what releases the
+      // new row count.
+      target_chunk->get_scoped_mvcc_data_lock()->grow_by(num_rows_for_target_chunk, context->transaction_id());
+
       // Grow data Segments.
+      // Do so in REVERSE column order so that the resize of `Chunk::_segments.front()` happens last. It is this last
+      // resize that makes the new row count visible to the outside world.
```

Comment:
Don't we need a fence here? Right now, there is nothing that keeps the compiler from resizing the segments in any order.

Reply:
I probably thought of tbb::concurrent_vector::resize() as a full-seq-consistency operation that doesn't need fences....

Added this after the column resize, but I have no idea whether it is necessary or not or whether a release-barrier would be enough...:

```c++

        // Make sure the first columns rewrite actually happens last and doesn't get reordered.
        std::atomic_thread_fence(std::memory_order_seq_cst);
```

Reply:
I feel like it has to be a seq_cst barrier to make sure the first column's new size only gets observed after the other resizes completed, but it feels awkward...

Reply:
Yes, the concurrent_vector should be fine for now, but once we put in std::vectors, I can almost guarantee that this will be forgotten.